src/sentry/api/serializers/models/releaseactivity.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from typing import Any, Mapping, MutableMapping, Sequence, TypedDict
+
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models import Group
+from sentry.models.releaseactivity import ReleaseActivity
+from sentry.types.releaseactivity import ReleaseActivityType
+from sentry.utils.json import JSONData
+
+
+class ReleaseActivityResponse(TypedDict):
+    id: int
+    type: str
+    data: Mapping[str, JSONData]
+    date_added: datetime
+
+
+@register(ReleaseActivity)
+class ReleaseActivitySerializer(Serializer):
+    def get_attrs(
+        self, item_list: Sequence[ReleaseActivity], user: Any, **kwargs: Any
+    ) -> MutableMapping[Any, Any]:
+        groups = list(
+            Group.objects.filter(
+                id__in=[
+                    i.data.get("group_id") for i in item_list if i.data and i.data.get("group_id")
+                ]
+            )
+        )
+        serialized_groups = {g["id"]: g for g in serialize(groups, user)}
+
+        def _expand_group(d: Mapping[str, Any]) -> Mapping[str, Any]:
+            if d.get("group_id"):
+                d_copy = dict(d)
+                d_copy["group"] = serialized_groups.get(str(d.get("group_id")), None)
```

Comment:
If the group has been deleted, should we still return these rows?

Reply:
Hmmm, didn't think of that. Would it make more sense to put that logic in the FE?

Reply:
If we had a relational link to groups from the table it'd be easy to filter these rows out. Not sure if we need to do that now,  but it might be better to store relational data outside of the json blob if possible

Reply:
I haven't looked into the semantics of how Group deletions work. Does that mark it as deleted, or does it actually remove the row entirely?

How would one know if a Group is deleted?

Tbh, I hadn't thought enough about the timeline and how much 'state' we should be reflecting there. A group deletion activity might be useful to have in the timeline.

Reply:
I see there's a GroupStatus here:
https://github.com/getsentry/sentry/blob/94c896a4a3663abbd31775957f1aa5448fde5491/src/sentry/models/group.py#L379-L388

But that doesn't include deletion.

Reply:
Groups are fully removed from the database after deletion, which is either explicit, or after 90 days of no new events.

Actually, I guess if we had a relation to the `Group` table then we'd just automatically have these rows removed once a group is deleted, which also solves the problem (unless you want to keep them... but there's no information about the group, so not sure how useful that is)

Reply:
Yeah, I dived a bit to see how deletion works and it does look like the rows are completely wiped out:
https://github.com/getsentry/sentry/blob/9ced05b0cf494b8398b5ce1c23957b620a8619ee/src/sentry/deletions/defaults/group.py#L100-L134

In the case where someone deletes the group, the returned data would be `group=None`. I think we could put in some logic in the FE to filter these cases out.